cookbook/basic_rag_example/pinecone_embeddings.py

```diff
@@ -0,0 +1,151 @@
+"""A script using RAG and Pinecone to summarize relevant articles from a large database.
+
+First we load in a dataset of news articles from 2004 and embed them. Then we use
+Pinecone to efficiently find the articles with most relevance to a query, then use said
+articles in a prompt to get the summary.
+"""
+import ast
+import os
+
+import pandas as pd
+import tiktoken
+from pinecone import Pinecone, ServerlessSpec
+
+from mirascope import OpenAIChat, Prompt
+
+os.environ["OPENAI_API_KEY"] = "YOUR_API_KEY"
+chat = OpenAIChat()
+embeddings_model = "text-embedding-ada-002"
+
+
+def load_data(url) -> pd.DataFrame:
+    """Loads data from a url, and splits larger texts into smaller chunks."""
+    df = pd.read_csv(url)
+    split_articles = []
+    encoder = tiktoken.encoding_for_model("gpt-3.5-turbo")
+
+    def split_text(text, tokens, max_tokens):
+        """Roughly splits a text into chunks of max_tokens size.
+
+        Note that tokens and characters do not have an exact correspondence, so in
+        certain edge cases a chunk may be slightly larger than max_tokens.
+        """
+        words = text.split()
+        num_splits = len(tokens) // max_tokens + 1
+        words_per_split = len(words) // num_splits
+        split_texts = [
+            " ".join(words[i : i + words_per_split])
+            for i in range(0, len(words), words_per_split + 1)
+        ]
+        return split_texts
+
+    for i, row in df.iterrows():
+        text = row["Text"]
+        tokens = encoder.encode(text)
+        if len(tokens) > 1000:
+            split_articles += split_text(text, tokens, 1000)
+            df.drop(i, inplace=True)
+
+    # Long texts which were dropped from the dataframe are now readded.
+    df = pd.concat(
+        [df, pd.DataFrame(split_articles, columns=["Text"])], ignore_index=True
+    )
+
+    return df
+
+
+def embed_data(df, batch_size):
+    """Embeds a dataframe of texts and stores it in a new column."""
+    embeddings = []
+    # ada 02 supports ~8000 tokens and 8 articles fit in the max batch size
```

Comment:
this comment is duplicated. I would suggest removing it here and leaving it below